packages/stg-evm-v2/src/StargateOFT.sol

```diff
@@ -22,6 +22,12 @@ contract StargateOFT is StargateBase {
         address _owner
     ) StargateBase(_token, IERC20Metadata(_token).decimals(), _sharedDecimals, _endpoint, _owner) {}
 
+    /// @notice Transfer ownership of the token to a new owner.
+    /// @param _newOwner The account to set as owner
+    function transferTokenOwnership(address _newOwner) external onlyOwner {
+        IERC20Minter(token).transferOwnership(_newOwner); // TODO how do i know this is the correct function to call -- where exactly does this lead after the IERC20Minter?
+    }
+
```

Comment:
I have 2 questions here:
1. See todo -- how do i know this is the correct function to call? Where exactly does this lead after the IERC20Minter? I feel like I would need to update another contract like [ERC20Token](https://github.com/stargate-protocol/stargate-v2/blob/main/packages/stg-evm-v2/src/mocks/ERC20Token.sol) (this one is just for testing tho), [OFTTokenERC20](https://github.com/stargate-protocol/stargate-v2/blob/main/packages/stg-evm-v2/src/utils/OFTTokenERC20.sol), or [OFTTokenERC20Upgradeable](https://github.com/stargate-protocol/stargate-v2/blob/main/packages/stg-evm-v2/src/utils/OFTTokenERC20Upgradeable.sol).
2. Is it okay to update `StargateOFT.sol` or should I create a new `StargateOFTUSDT.sol` that extends `StargateOFT.sol` and adds this `transferTokenOwnership` function? I updated `StargateOFT.sol` for now thinking it may avoid downstream changes to deploy/config scripts, but please let me know if this is not recommended! I made it `onlyOwner` for added security as well

Reply:
1) You would simply test that it works, and then you know its fine
2) Im leaning towards just implementing it for all tokens, and it will just fail for non usdt ones

Reply:
Also you can use Ownable(token) instead of IErc20Minter

Reply:
Amazing, thank you both! Just tested this out after deploying USDT and stargate on testnet (and configuring) and it works like a charm!